jekyll/_cci2/language-go.md

```diff
@@ -75,69 +112,119 @@ jobs:
           command: |
-            set -eu
-            go get -t -d -v ./...
-            export DATABASE_URL="postgres://ubuntu@localhost:5432/service_test?sslmode=disable"
-            go test -v -race ./... | go-junit-report > /tmp/test-results/unit-tests.xml
+            trap "go-junit-report <${TEST_RESULTS}/go-test.out > ${TEST_RESULTS}/go-test-report.xml" EXIT
+            make test | tee ${TEST_RESULTS}/go-test.out
+```
+
+Our project uses `make` for building and testing (you can see `Makefile` [here](https://github.com/circleci/cci-demo-go/blob/master/Makefile)), so we can just run `make test`. In order to collect test results and upload them later (read more about [test results]({{ site.baseurl }}/2.0/project-walkthrough/#store-test-results)) we have to use `go-junit-report`, so we can do it like this:
+
+``` YAML
+make test | go-junit-report > ${TEST_RESULTS}/go-test-report.xml
+```
+
+However in this case all output from `make test` will go straight into `go-junit-report` without appearing in `stdout`. We can solve this by using two standard Unix commands `tee` and `trap`. The first one allows to duplicate output into `stdout` and somewhere else ([read more](http://man7.org/linux/man-pages/man1/tee.1.html)). The second one allows to specify some command to be executed on script exit ([read more](http://man7.org/linux/man-pages/man1/trap.1p.html)). So we can do:
+
+``` YAML
+trap "go-junit-report <${TEST_RESULTS}/go-test.out > ${TEST_RESULTS}/go-test-report.xml" EXIT
+make test | tee ${TEST_RESULTS}/go-test.out
 ```
 
+Now then we know that our unit tests succeeded we can start our service and validate it's running.
+
+``` YAML
+      - run:
+          name: Build service
+          command: make
+
+      - run:
+          name: Start service
+          environment:
+            CONTACTS_DB_URL: "postgres://ubuntu@localhost:5432/contacts?sslmode=disable"
+            CONTACTS_DB_MIGRATIONS: /go/src/github.com/circleci/cci-demo-go/db/migrations
+          command: ./workdir/contacts
+          background: true
+
+      - run:
+          name: Validate service is working
+          command: curl --retry 10 --retry-delay 1 --retry-connrefused http://localhost:8080/contacts/test
+```
+
+To start service we need to build it first. After that we use the same environment variables as we did in the testing step for service to start. We're using `background: true` to keep service running and proceed to the next step where we use `curl` to validate service successfully started and responding to our request.
+
 Finally, let's specify a path to store the results of the tests.
 
 ```YAML
-...
       - store_test_results:
           path: /tmp/test-results
 ```
 
-Nice! You just set up CircleCI for a Go app. Check out our [project’s build page](https://circleci.com/gh/circleci/cci-demo-go).
+Nice! You just set up CircleCI 2.0 for a Go app. Check out our [project’s build page](https://circleci.com/gh/circleci/cci-demo-go).
```

Comment:
@ndintenfass mentioned that we should be writing these docs as if there is no distinction between a 1.0 and 2.0 version of CircleCI.
I think that's why I originally removed "2.0" from this paragraph.

I won't keep swapping things back until everyone's aligned, though. :)